cfgov/filing_instruction_guide/models.py

```diff
@@ -12,16 +14,43 @@
 from wagtail.admin.forms import WagtailAdminPageForm
 from wagtail.core.fields import StreamField
 
+import requests
+from filing_instruction_guide import import_data_points
 from filing_instruction_guide.blocks import (
     FigLevel3Subsection,
     FigSection,
     FigSubsection,
 )
+from modelcluster.models import ClusterableModel
 
 from v1.models.base import CFGOVPage
 
 
 class FIGPageForm(WagtailAdminPageForm):
+    def clean(self):
+        data = super().clean()
+        field = "data_points_download_location"
+        try:
+            import_data_points.run(data, self.instance)
+        except KeyError as err:
+            msg = f"""
+            The JSON file provided does not match the expected format.
+            Missing key: {err}
+            """
+            self.add_error(field, forms.ValidationError(msg))
+        except json.JSONDecodeError:
+            msg = """
+            Unable to parse the input file as JSON.
+            Please check the url and the format of the file.
+            """
+            self.add_error(field, forms.ValidationError(msg))
+        except requests.exceptions.RequestException:
+            msg = """
+            The file could not be downloaded at the specified URL.
```

Comment:
When I pass it a [malformed file](https://GHE/raw/CFPB/1071-fig-content/master/test_data_points/malformed.json) (there's an extra curly brace on line 3) it raised this generic exception (`The file could not be downloaded at the specified URL.`) instead of a `JSONDecodeError`.

Is it possible to return some sort of `Your JSON is borked` message? I anticipate users will often provide invalid JSON because it's easy to include erroneous brackets and quotes.

Reply:
Good call. I was catching the JSON error incorrectly. Fixed in 485f794.